pkg/controllers/disruption/events/events.go

```diff
@@ -121,6 +121,18 @@ func Blocked(node *v1.Node, nodeClaim *v1beta1.NodeClaim, reason string) []event
 	}
 }
 
+func NodePoolBlockedForDisruptionReason(nodePool *v1beta1.NodePool, reason v1beta1.DisruptionReason) events.Event {
+	return events.Event{
+		InvolvedObject: nodePool,
+		Type:           v1.EventTypeNormal,
+		Reason:         "DisruptionBlocked",
+		Message:        fmt.Sprintf("No allowed disruptions for disruption reason %s due to blocking budget", reason),
+		DedupeValues:   []string{string(nodePool.UID), string(reason)},
+		// Set a small timeout as a NodePool's disruption budget can change every minute.
```

Comment:
1 minute isn't accurate, it can change every 10 seconds right? (Each disruption controller.Disrupt() loop)